src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadata.scala

```diff
@@ -116,14 +116,21 @@ object DatasetMetadata {
     }
   }
 
-  trait RequiresNonEmpty {
-    def requireNonEmptyString[T](value: T, tag: String): Unit = {
+  trait Requirements {
```

Comment:
Why not extract this trait into its own class. Especially since you use it in so many other classes.